fw/http_parser.c

```diff
@@ -5860,8 +5913,22 @@ __h2_req_parse_cache_control(TfwHttpReq *req, unsigned char *data, size_t len,
 
 	__FSM_START(parser->_i_st);
 
+	parser->cache_control.dir_flag = 0;
+
+	__FSM_STATE(Req_I_CC_start) {
+		if (IS_WS(c))
+			__FSM_I_MOVE(Req_I_CC_start);
```

Comment:
yeah, it seems we had the issue in the parsing function because it didn't process WS at the beginning. However, it seems we can move the check and transition to `Req_I_CC`